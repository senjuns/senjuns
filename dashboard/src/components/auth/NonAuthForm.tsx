import { FC, ReactNode } from 'react';
import styled from 'styled-components';

import { ScreenSize, Sizes } from '../../shared/constants';

interface NonAuthFormProps {
  /**
   * Header of the NonAuthForm
   */
  header?: ReactNode;
  /**
   * Instruction Header of the NonAuthForm. like `Reset Password` and `Forgot Password`
   */
  instructionHeader?: ReactNode;
  /**
   * Instructions of the NonAuthForm.
   */
  instructions?: ReactNode;
  /**
   * Link of the NonAuthForm.
   */
  footer?: ReactNode;
}

/**
 * NonAuthForm presentation component.
 *
 * @param {NonAuthFormProps} - The component props.
 * @returns {JSX.Element} - The rendered component.
 */
export const NonAuthForm: FC<NonAuthFormProps> = ({
  children,
  header,
  instructionHeader,
  instructions,
  footer,
}) => {
  return (
    <>
      {header && <Header>{header}</Header>}
      {instructionHeader && (
        <InstructionHeader>{instructionHeader}</InstructionHeader>
      )}
      {instructions && <Instructions>{instructions}</Instructions>}
      {children}
      <Footer>{footer}</Footer>
    </>
  );
};

const Header = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 100px;

  @media only screen and (max-width: ${ScreenSize.sm}px) {
    display: none;
  }
`;

const InstructionHeader = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
`;

const Instructions = styled.div`
  margin-bottom: 100px;

  @media only screen and (max-width: ${ScreenSize.sm}px) {
    margin-bottom: 50px;
  }
`;

const Footer = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 20px;
  gap: ${Sizes.xxSmall}px;
`;
