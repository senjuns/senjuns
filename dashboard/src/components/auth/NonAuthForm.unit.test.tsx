// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { NonAuthForm } from './NonAuthForm';

describe('<NonAuthForm />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <NonAuthForm header="Header" instructions="Instructions" footer="Footer">
        Form Values
      </NonAuthForm>
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
