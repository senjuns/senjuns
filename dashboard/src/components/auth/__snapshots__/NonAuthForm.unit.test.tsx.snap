// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NonAuthForm /> renders correctly. 1`] = `
<React.Fragment>
  <styled.div>
    Header
  </styled.div>
  <styled.div>
    Instructions
  </styled.div>
  Form Values
  <styled.div>
    Footer
  </styled.div>
</React.Fragment>
`;
